### ISSUE TITLE
Trying to play a video crashes Kodi

### ISSUE THREAD

**Issue by jsteel44** [asciidisco/plugin.video.netflix#280]
*I'm submitting a ...*
  - [x] bug report

## General infomration
Trying to play a video crashes Kodi

### Prerequisites

* [x] Are you running the latest version? Yes

### Description
Everything looks good up until trying to play a Video. Sometimes I see a loading icon, sometimes I see a black screen, sometimes I see a black screen and then it goes back to the listing and freezes until it crashes moments later.

### Steps to Reproduce

1. Log in to Netflix (select Netflix user)
2. Select any video to play
3. After about 10 to 30 seconds, Kodi restarts

**Expected behavior:** The video should play

**Actual behavior:** Kodi crashes/restarts

### Installation
* [x] I installed the plugin via zip from the Releases page

### Operating System
* [x] Linux (ARM)

#### Additional informatin on the environment
LibreELEC 18.0-ALPHA1 Git: 69a6725. With /storage/.kodi/cdm/libwidevinecdm.so 

## Debug log
https://bpaste.net/show/985e1dd705ff

**Comment by Jooosi** [comment]
Probably you somehow updated your libwidevinecdm.so, try to delete the libwidevinecdm.so from "...\userdata\addon_data\script.module.inputstreamhelper\cdm" and reinstall it again using the known SSH method:
1. Run `wget http://nmacleod.com/public/libreelec/getwidevine.sh`
2. Run `sh getwidevine.sh`

Had the same problem and these steps worked for me.

**Comment by tacrjpa** [comment]
Hi @Jooosi, i have a raspberry pi 3 with kodi 18 and also updated widevine in an attempt to play Amazon VOD.  Netflix then crashed Kodi when playing a video. I've tried do reinstall with your tip:

    Run wget http://nmacleod.com/public/libreelec/getwidevine.sh
    Run sh getwidevine.sh

But  got:

> _ERROR: Unable to determine ARCH for this system!
> ERROR: Specify ARCH=arm or ARCH=x86_64 if /etc/os-release does not define a suitable ARCH_

Then i tried to specify ARCH=arm in /etc/os-release and got:

> _Detected ARCH: ARCH=arm
> Download directory: /home/osmc/.kodi/cdm
> ERROR: Arch ARCH=arm is not supported!_

Had you met this problem by chance?

And thank you in advance!

**Comment by Jooosi** [comment]
@tacrjpa No, I did not have this problem. I'm running an OpenELEC system, not OSMC. Perhaps you should change to OpenELEC and the actual Milhouse build, the bug with the updated widevine crashing Kodi was fixed in build 0207.

**Comment by MilhouseVH** [comment]
@tacrjpa I'd say you've now messed up your `/etc/os-release` file, possibly by adding `ARCH=ARCH=arm`. You should not have edited `/etc/os-release`, and doing so may cause you other unrelated problems.

As per the message you were given, you should have specified `ARCH=arm` when running the script, ie.:
```
curl -sL http://nmacleod.com/public/libreelec/getwidevine.sh | ARCH=arm sh
```

However, the the version of `libwidevinecdm.so` installed by the script is no longer compatible with Amazon after they changed their licence API (or something) and you will need to install a more recent version from an alternative source. I no longer update `getwidevine.sh` as it has served its purpose and inputstream helper addons should now fill the gap, updating `libwidevinecdm.so` whenever necessary.

The latest version of `libwidevinecdm.so` does introduce a new issue that will crash Kodi when playing Netflix and Amazon streams, and this is only fixed in my LibreELEC test builds starting with [#0207](http://forum.kodi.tv/showthread.php?tid=298461&pid=2701351#pid2701351).

OpenELEC is no longer updated/maintained as far as anyone can tell, so you definitely don't want to go near it.

tl;dr: 
1. Roll back your `/etc/os-release` change
2. Use an inputstream helper add-on to install the latest `libwidevinecdm.so`
3. Update to LibreELEC test build #0207 or later

**Comment by Jooosi** [comment]
Oh sorry, of course I meant LibreELEC, not OpenELEC. :)

**Comment by liberty-developer** [comment]
Can we close this issue report?

**Comment by asciidisco** [comment]
Closing this as widevine is installed automatically & there are no further reports.
